connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -135,8 +161,7 @@ public class KafkaBasedLogTest {
     @SuppressWarnings("unchecked")
     @Before
     public void setUp() {
-        store = PowerMock.createPartialMock(KafkaBasedLog.class, new String[]{"createConsumer", "createProducer"},
-                TOPIC, PRODUCER_PROPS, CONSUMER_PROPS, consumedCallback, time, initializer);
+        store = new MockedKafkaBasedLog(TOPIC, PRODUCER_PROPS, CONSUMER_PROPS, () -> null, consumedCallback, time, null);
```

Comment:
Minor tidbit, in the old test `initializer` was actually the wrong type (its a `Runnable` when it should be a `Supplier`) so if it was ever accessed during a test it would have thrown a `ClassCastException` (or something along those lines). Furthermore the `createPartialMock` was actually missing an argument.

This is why the version in this PR as `() -> null` as a `topicAdminSupplier` and `null` for the `initializer`. These fields are never actually called (or if they are they are setup with another method, i.e. `setupWithAdmin` which inserts the correct fields)

Reply:
> I think the @Mock private Runnable initializer is the incorrect signature here, and should be changed to be a Consumer mock. Otherwise the initializer value isn't behaving like a mock for the mocked kafka based log.

So it appears that the `@Mock private Runnable initializer` is being used correctly in the `setupWithAdmin` method and if you look at `setupWithAdmin` you can confirm its correct because we aren't using a mock, instead we actually using the raw constructor, i.e.

```java
    private void setupWithAdmin() {
        Supplier<TopicAdmin> adminSupplier = () -> admin;
        java.util.function.Consumer<TopicAdmin> initializer = admin -> { };
        store = new MockedKafkaBasedLog(TOPIC, PRODUCER_PROPS, CONSUMER_PROPS, adminSupplier, consumedCallback, time, initializer);
    }
```

I believe the oversight was just in the `setUp` of the mock.

Reply:
See https://github.com/apache/kafka/pull/12781#discussion_r1006616025 for a solution